homeassistant/components/hunterdouglas_powerview/config_flow.py

```diff
@@ -36,44 +37,72 @@ async def validate_input(hass: core.HomeAssistant, hub_address: str) -> dict[str
 
     try:
         async with asyncio.timeout(10):
-            device_info = await async_get_device_info(pv_request, hub_address)
+            hub = Hub(pv_request)
+            await hub.query_firmware()
+            if hub.role != "Primary":
+                raise UnsupportedDevice(
+                    f"{hub.name} ({hub.hub_address}) is performing role of {hub.role} Hub. Only the Primary Hub can manage shades"
+                )
+            device_info = await async_get_device_info(hub)
+    except UnsupportedDevice as unsupported_err:
+        _LOGGER.debug(unsupported_err)
+        raise UnsupportedDevice from unsupported_err
```

Comment:
I think we want to trap this in `_async_validate_or_error` instead and change the error string returned so we can add it to `strings.json`
```suggestion
```